tidy3d/plugins/mode/solver.py

```diff
@@ -525,6 +525,7 @@ def set_initial_vec(Nx, Ny, mat_dtype=np.complex128, is_tensorial=False):
 
     # Initialize the vector
     size = (Nx, Ny, len_multiplier)
+    np.random.seed(0)
```

Comment:
One problem is that this will not affect random generation used internally inside numpy/scipy functions. I think this is not a problem here, because we actually have our own `set_initial_vec` function. However, if we didn't have that, and wanted to seed the in-built `scipy.linalg.eigs` initial vector generation, I don't know how to do it correctly if not with `np.random.seed(0)`.

Because of this I feel like we should stick with `np.random.seed(0)` to make sure that both our functions and internal functions work deterministically?

Reply:
Ah, yeah, makes sense. I looked at the source code of `scipy.linalg.eigs` out of curiosity. Turns out, it relies on fortran library arpack which internally generates initial random vector if it is not provided https://github.com/opencollab/arpack-ng/blob/master/SRC/sgetv0.f. Other than that it seems there is no other randomness in the eigenvalue solver